### PR TITLE
Added some basic documentation

### LINKED ISSUE #3: documentation?
the readme has a bullet point "Slugifier documentation" which is not linked to anywhere. we should provide a bit of documentation. given the nature of this thing as simply a contract hopefully to be replaced by a PSR one day, we could maybe simply include the documentation in the README and link from the cmf docs to the README in the places where people need that information.

### PR DESCRIPTION
**Pull request by dantleech**
This PR adds some basic documentation.

fix #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,28 +7,65 @@
...
-The Slugifier API provides a basic interface to use for third party slugifiers.
-Besides this, it comes with a usefull `CallbackSlugifier` which is capable of
-handling most third party slugifiers.
+A "slugifier" is a function which transforms a string `such as this` into a
+URL-friendly string `such-as-this`. Slugifiers are also known as "urlizers".
 
+This package does not contain a slugifier itself, but provides a standard
+interface to use for third party slugifiers.  Besides this, it comes with a
+useful `CallbackSlugifier` which is capable of handling most third party
+slugifiers.
```
Comment (dbu) [inline README.md]: which is capable of wrapping most third party slugifiers to the SlugifierInterface provided by this package.

**Comment by dbu** [comment]
:+1: 

should we mention FIG somewhere? maybe a disclaimer like "Ideally this package would be a PSR standard. We [proposed this to FIG](https://groups.google.com/forum/?fromgroups=#!topic/php-fig/J-6s9Wlyk-A) but unfortunately the proposal did not get enough interest. We would still be happy to contribute this to a PSR should the interest come up and deprecate this package in favor of the PSR one."

**Comment by wouterj** [comment]
:+1:

**Comment by wouterj** [comment]
Fyi, I made some minor changes in 1334b36d83195fc0b2b194ac8d46d61c6d88c080

**Pull request merged**